[inductor] Allow `tensors` kwarg in sink_cat_after_pointwise

bertmaher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #97028
* __->__ #97019

Lacking handling of kwargs strikes again.

Differential Revision: [D44166740](https://our.internmc.facebook.com/intern/diff/D44166740/)

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

bertmaher committed:
[inductor] Allow `tensors` kwarg in sink_cat_after_pointwise

Lacking handling of kwargs strikes again.

Differential Revision: [D44166740](https://our.internmc.facebook.com/intern/diff/D44166740/)

[ghstack-poisoned]

jansel commented on torch/_inductor/overrides.py:
```diff
@@ -299,13 +316,17 @@ def is_pointwise_unary(node):
 
         if user and is_pointwise_unary(user):
             with g.inserting_before(node):
+                norm_cat = NormalizedCatNode(node)
```
```suggestion
                def cat_args(tensors, dim):
                   return tensors, dim
                tensors, dim = cat_args(*node.args, **node.kwargs)
```

bertmaher replied:
Oh that's cute haha.  I'll take it!

jansel reviewed (approved):
Yet another reason to do optimizations after AotAutograd (which has normalized args for you already)

bertmaher committed:
shorten normalization on "[inductor] Allow `tensors` kwarg in sink_cat_after_pointwise"


Lacking handling of kwargs strikes again.

Differential Revision: [D44166740](https://our.internmc.facebook.com/intern/diff/D44166740/)

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

bertmaher committed:
black on "[inductor] Allow `tensors` kwarg in sink_cat_after_pointwise"


Lacking handling of kwargs strikes again.

Differential Revision: [D44166740](https://our.internmc.facebook.com/intern/diff/D44166740/)

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

bertmaher commented:
And yes, I want to move this (and maybe other transforms) after AOTAutograd for that reason.  I'll take a swing at it next week.

ngimel commented:
But don't you want AOTAutograd to run on a transformed graph so that backward is also benefiting from this?

jansel commented:
That is correct.  Maybe we would need a pattern for the backwards as well?

bertmaher committed:
rebase on "[inductor] Allow `tensors` kwarg in sink_cat_after_pointwise"


Lacking handling of kwargs strikes again.

Differential Revision: [D44166740](https://our.internmc.facebook.com/intern/diff/D44166740/)

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

bertmaher committed:
Update on "[inductor] Allow `tensors` kwarg in sink_cat_after_pointwise"


Lacking handling of kwargs strikes again.

Differential Revision: [D44166740](https://our.internmc.facebook.com/intern/diff/D44166740/)

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

bertmaher committed:
lint on "[inductor] Allow `tensors` kwarg in sink_cat_after_pointwise"


Lacking handling of kwargs strikes again.

Differential Revision: [D44166740](https://our.internmc.facebook.com/intern/diff/D44166740/)

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

bertmaher committed:
rebase on "[inductor] Allow `tensors` kwarg in sink_cat_after_pointwise"


Lacking handling of kwargs strikes again.

Differential Revision: [D44166740](https://our.internmc.facebook.com/intern/diff/D44166740/)

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

bertmaher commented:
@bertmaher has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D44166740).
